Add conda recipe for sed-plume

mcflugen opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/sed-plume```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/sed-plume```) and found some lint.

Here's what I've got...


For **recipes/sed-plume**:

 * Non noarch packages should have python requirement without any version constraints.
 * Non noarch packages should have python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/sed-plume```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/sed-plume```) and found some lint.

Here's what I've got...


For **recipes/sed-plume**:

 * The build section contained an unexpected subsection name. patches is not a valid subsection name.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/sed-plume```) and found it was in an excellent condition.

mcflugen commented:
@conda-forge/help-python This is ready for review. Thanks!

mcflugen commented:
@conda-forge/help-python Sorry for the false alarm. This ready for review now—for reals. Thanks!

mcflugen commented:
@conda-forge/help-python Please let me know if there is anything I can do to move this along. Thanks!

carterbox commented on recipes/sed-plume/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+  run:
+    - python
+    - gsl
+    - libopenblas
```
```suggestion
```
BLAS is not a python library or a binary. It should not be in the run requirements without also being in the host section. If your package links against BLAS, please follow the BLAS directions in the docs.

https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=blas#blas

But also, does this package link to BLAS? The linker doesn't detect links to BLAS on any platform.
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=529045&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=4853

mcflugen replied:
@carterbox I had added *libopenblas* as a requirement to solve a linker error I was getting but that error seems to have gone away!

carterbox replied:
Looks like your project includes the following headers

```
#include "gsl/gsl_cblas.h"
#include "gsl/gsl_blas_types.h"
#include "gsl/gsl_blas.h"
```
And the logs show that `centerline.cpython-39-x86_64-linux-gnu.so` links to libopenblas.

However, I don't know enough about your package to tell whether this link to libopenblas is because there are actually libopenblas symbols included in those headers or if there is just a mistake in your build process that accidentally links to libopenblas when it is not needed.

mcflugen replied:
Interesting. I also would have thought that if *libopenblas* was required it would have been listed as a dependency for *gsl*.

carterbox commented on recipes/sed-plume/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+    - scipy
+    - tomlkit
+
+test:
+  imports:
+    - plume
+  commands:
+    - plume --version
+    - plume --help
+    - plume generate --help
+    - plume run --help
```
```suggestion
test:
  requires:
    - pip
  imports:
    - plume
  commands:
    - plume --version
    - plume --help
    - plume generate --help
    - plume run --help
    - pip check
```

mcflugen commented:
@carterbox I added *netcdf4* as a requirement to fix the *pip check* error. *landlab* no longer uses *netcdf* directly but instead uses it through *xarray*. I'm not sure why the *netcdf* library is no longer coming with *xarray*.

carterbox reviewed (changes requested):
This package is installing a "tests" module directly to site-packages. This is probably an error. You will need to patch the upstream package to not include tests in the global namespace.

Probably, you should add an exclude statement to this find_package config.

https://github.com/mcflugen/plume/blob/66b4c80517891a14abe1bed781adf8415d4cfb2e/pyproject.toml#L80

mcflugen commented on recipes/sed-plume/meta.yaml:
```diff
@@ -0,0 +1,70 @@
+    - python
+    - gsl
+    - landlab >=2
+    - netcdf4
```
@carterbox *landlab* no longer uses *netcdf4* directly (i.e. it doesn't include anything like `import netcdf4`). Instead, it uses *xarray* for reading and writing to *netCDF* files. However, as you say, *xarray* may in the future change its implementation of its netcdf io functions to not use the *netcdf4* package so I don't think it should include it in its dependencies. In fact, I'm now not certain *xarray* does use *netcdf4*.

mcflugen replied:
Ah, never mind. I see that *netcdf4* is an optional dependency for *xarray that's required only if you want to read/write netcdf files.

mcflugen commented:
@carterbox Do you think this is ready to go now? I've patched *landlab* so that it doesn't install its tests in *site-packages*, fixed its dependencies to include *netcdf4*, and added `pip check` to *sed-plume*'s tests.

mcflugen commented:
@conda-forge/help-python I think this is ready to go now. Thanks!

mcflugen commented:
@conda-forge/staged-recipes I fear my pull request may have gotten lost in the shuffle. Anyway, I think it's is ready to go now. Thanks!
